crates/rb-sys/build.rs

```diff
@@ -128,13 +128,54 @@ fn main() {
         .header("wrapper.h")
         .use_core()
         .ctypes_prefix("::libc")
-        .allowlist_file(".*ruby.*")
+        .allowlist_function("^(onig(enc)?|rb|ruby)_.*")
+        .allowlist_function("eaccess")
+        .allowlist_function("explicit_bzero")
+        .allowlist_function("setproctitle")
+        .allowlist_type("VALUE")
+        .allowlist_type("Regexp")
+        .allowlist_type("^(Onig|R[A-Z]|re_|rb_|rbimpl_|ruby_|st_).*")
+        .allowlist_var("^(Onig|rb_|ruby_).*")
+        .allowlist_var("^(FMODE_|INTEGER_|HAVE_|ONIG|Onig|RBIMPL_|RB_|RGENGC_|RUBY_|SIGNEDNESS_|SIZEOF_|USE_).*")
+        .allowlist_var("ATAN2_INF_C99")
+        .allowlist_var("BROKEN_BACKTRACE")
+        .allowlist_var("BROKEN_CRYPT")
+        .allowlist_var("CASEFOLD_FILESYSTEM")
+        .allowlist_var("COROUTINE_H")
+        .allowlist_var("DLEXT")
+        .allowlist_var("DLEXT_MAXLEN")
+        .allowlist_var("ENUM_OVER_INT")
+        .allowlist_var("FALSE")
+        .allowlist_var("INCLUDE_RUBY_CONFIG_H")
+        .allowlist_var("INTERNAL_ONIGENC_CASE_FOLD_MULTI_CHAR")
+        .allowlist_var("LIBDIR_BASENAME")
+        .allowlist_var("NEGATIVE_TIME_T")
+        .allowlist_var("PATH_ENV")
+        .allowlist_var("PATH_SEP")
+        .allowlist_var("POSIX_SIGNAL")
+        .allowlist_var("STACK_GROW_DIRECTION")
+        .allowlist_var("STDC_HEADERS")
+        .allowlist_var("ST_INDEX_BITS")
+        .allowlist_var("THREAD_IMPL_H")
+        .allowlist_var("THREAD_IMPL_SRC")
+        .allowlist_var("TRUE")
+        .allowlist_var("UNALIGNED_WORD_ACCESS")
+        .allowlist_var("UNLIMITED_ARGUMENTS")
+        .allowlist_var("_ALL_SOURCE")
+        .allowlist_var("_GNU_SOURCE")
+        .allowlist_var("_POSIX_PTHREAD_SEMANTICS")
+        .allowlist_var("_REENTRANT")
+        .allowlist_var("_TANDEM_SOURCE")
+        .allowlist_var("_THREAD_SAFE")
+        .allowlist_var("__EXTENSIONS__")
+        .allowlist_var("__STDC_WANT_LIB_EXT1__")
         .rustified_enum("*")
         .blocklist_item("ruby_abi_version")
         .derive_eq(true)
         .derive_debug(true)
         .clang_args(clang_args)
         .parse_callbacks(Box::new(bindgen::CargoCallbacks))
+        .generate_comments(false)
```

Comment:
Curious as to your thoughts on this, were comments causing an issue? Are they just not helpful?

Reply:
The comments for Ruby < 3.1 are pretty much worthless, but 3.1 added pretty nice documentation comments to the headers, they just don't translate across great.

I think when I first looked I happened to hit on one that was completely broken, but now I look again they are badly formatted, but I guess useable:
<img width="869" alt="Screenshot 2022-04-18 at 21 11 50" src="https://user-images.githubusercontent.com/37348/163919089-2166b49e-9bb3-49fe-955a-a7fc02be9e4f.png">

I can drop this change if you prefer.

Reply:
My vote would be to keep it for now, since it's at least somewhat readable on 3.1